public/app/plugins/datasource/elasticsearch/query_builder.ts

```diff
@@ -318,6 +318,11 @@ export class ElasticQueryBuilder {
         metricAgg = { field: metric.field };
       }
 
+      if (metric.inlineScript) {
```

Comment:
Just to be functionally equivalent to the previous version, what do you think about adding this?

```suggestion
      const supportsInlineScript = !!queryDef.metricAggTypes.find(metricDef => metricDef.value === metric.type)
        ?.supportsInlineScript;

      if (supportsInlineScript && metric.inlineScript) {
```

the reason is that when changing metric type from the query editor we don't erase previously set properties from `settings`, which means you can set a script on a metric that supports it, then change the metric type to something that doesn't support it and the `inlineScript` would maintain the previous value. without this extra check the query builder would add `script: { 'inline/source': ...}` to every metric instead to only those that supports it.